Mathlib/Analysis/Convex/Radon.lean

```diff
@@ -48,3 +58,126 @@ theorem radon_partition (h : ¬ AffineIndependent 𝕜 f) :
+    (h_inter : ∀ I : Finset ι, I.card ≤ finrank 𝕜 E + 1 → (⋂ i ∈ I, F i).Nonempty) :
+    (⋂ i : ι, F i).Nonempty := by
+  by_cases h_card : Fintype.card ι < finrank 𝕜 E + 1
+  · exact helly_theorem_corner (le_of_lt h_card) h_inter
+  · generalize hn : Fintype.card ι = n
+    rw [not_lt, hn] at h_card
+    induction' n, h_card using Nat.le_induction with k h_card hk generalizing ι
+    · exact helly_theorem_corner (le_of_eq hn) h_inter
+    · /- Construct a family of vectors indexed by `ι` such that the vector corresponding to `i : ι`
+      is an arbitrary element of the intersection of all `F j` except `F i`. -/
+      let a (i : ι) : E := Set.Nonempty.some (s := ⋂₀ (F '' (Finset.univ.erase i))) <| by
+        let ι' :=  Finset.univ.erase i
+        let F' : ι' → Set E := fun i ↦ F i
+        rw [show ⋂₀ (F '' ι') = ⋂ i, F' i by simp [Set.iInter_subtype]]
+        apply hk (F := F')
+        · exact fun i ↦ h_convex ↑i
+        · intro J hJ_card
+          rw [show ⋂ i ∈ J, F' i = ⋂ i ∈ Finset.image Subtype.val J, F i by
+            simp [Set.iInter_subtype]]
+          exact h_inter _ (le_trans Finset.card_image_le hJ_card)
+        · simp [ι']; omega
+
+      /- This family of vectors is not affine independent because the number of them exceeds the
+      dimension of the space. -/
+      have h_ind : ¬AffineIndependent 𝕜 a := by
+        rw [← finrank_vectorSpan_le_iff_not_affineIndependent 𝕜 a (n := (k - 1))]
+        · exact (Submodule.finrank_le (vectorSpan 𝕜 (Set.range a))).trans (Nat.le_pred_of_lt h_card)
+        · omega
+      /- Use `Convex.radon_partition` to conclude there is a subset `I` of `ι` and a point `p : E`
+      which lies in the convex hull of either `a '' I` or `a '' Iᶜ`. We claim that `p ∈ ⋂₀ F`. -/
+      obtain ⟨I, p, h4_I, h4_Ic⟩ := Convex.radon_partition h_ind
+      use p
+      apply Set.mem_iInter_of_mem
+      intro i
+      /- It suffices to show that for any set `F i` in a subcollection `I` of `ι`, that the convex
+      hull of `a '' Iᶜ` is contained in `F i`. -/
+      suffices ∀ I : Set ι, i ∈ I → (convexHull 𝕜) (a '' Iᶜ) ⊆ F i by
+        by_cases h : i ∈ I
+        · exact this I h h4_Ic
+        · apply this Iᶜ h; rwa [compl_compl]
+      /- Given any subcollection `I` of `ι` containing `i`, because `F i` is convex, we need only
+      show that `a j ∈ F i` for each `j ∈ Iᶜ`. -/
+      intro I hi
+      rw [Convex.convexHull_subset_iff (h_convex i)]
+      rintro - ⟨j, hj, rfl⟩
+      /- Since `j ∈ Iᶜ` and `i ∈ I`, we conclude that `i ≠ j`, and hence by the definition of `a`:
+      `a j ∈ ⋂ F '' (Set.univ \ {j}) ⊆ F i`. -/
+      apply Set.mem_of_subset_of_mem (s₁ := ⋂₀ (F '' (Finset.univ.erase j)))
+      · apply sInter_subset_of_mem
+        use i
+        simp only [coe_erase, coe_univ, mem_diff, Set.mem_univ, mem_singleton_iff, true_and,
+          and_true]
+        exact fun h' ↦ hj (h' ▸ hi)
+      · apply Set.Nonempty.some_mem
+
+/-- The set version of `Convex.helly_theorem`. -/
```

Comment:
Can you avoid making the only description of an important "It's like this other result except for that thing that changes"? It makes it really annoying when you hover over it and try to understand what it says.
```suggestion
/-- **Helly's theorem** for finite sets of convex sets.

If `F` is a finite set of convex sets in a vector space of finite dimension `d`,
and any `d + 1` sets of `F` intersect, then all sets of `F` intersect. -/
```